sos/policies/distros/redhat.py

```diff
@@ -254,6 +254,14 @@ def get_upload_url(self):
             return self.commons['cmdlineopts'].upload_url
         if self.commons['cmdlineopts'].upload_protocol == 'sftp':
             return RH_SFTP_HOST
+        if self.commons['cmdlineopts'].upload_protocol == 's3':
+            endpoint = self.get_upload_s3_endpoint()
+            bucket = self.get_upload_s3_bucket()
+            if self.commons['cmdlineopts'].case_id:
+                rh_case_api = "/support/v1/cases/%s/attachments"
+                return f"{endpoint}/{bucket}" + rh_case_api % self.case_id
```

Comment:
Let's make this all f-string formatting, mixing the two syntax is a little awkward.

Reply:
I'll adjust this accordingly once I have additional feedback to https://github.com/sosreport/sos/pull/3891#issuecomment-2561360210